### ISSUE TITLE
Customization points are conflating concepts

### ISSUE THREAD

**Issue by tomaszkam** [atomgalaxy/review-executor-sendrecv#4]
The current definitions of customization points are conflating the concepts, by providing a fallback wrapping.  For example:

1. `executor` is `scheduler`:
The `exeuction::schedule` has a fallback defintion for the `executor`, that wraps int into void-sender. That means that for every `executor e` the `execution::schedule(e)` is well-formed, as consequence every `executor` is `scheduler` (both are `copy_constructible` and `equality_comparable`), but the `executor` does not subsume `scheduler`.

2. `executor` is `void-sender`, `void-sender` is `executor`:
The fallbacks in `connect` automatically wraps executor in sender, but again executor does not subsume `void_sender`.
Also `execution::execute` works with any `void_sender`, so `copy_constructible` and `equality_comparable` sender is executor.

The whole definition is very confusing and make it very hard to understand the difference and need for each concept. 

Even in the paper, the `sender_traits<S>` have as specialization for `void_sender` that is enabled if `executor-of-impl<S,as-invocable<void-receiver, S>>`, which imposes  `copy_constructible` and `equality_comparable` requirements, instead of `sender_to<S, void-receiver>`. This means that this specialization is enabled for `copy_constructible` senders (because they are `executors`), but not for normal senders.

**Comment by ericniebler** [comment]
I agree it is confusing, but I don't think it is a defect. Although a related problem is that an `executor` is made to satisfy `void-sender`, but really it more closely models a `scheduler`. The CPOs should be reworked to manifest that analog. Then you'll have executors that can be used as schedulers and _vice versa_.

We can't have the `executor` and `scheduler` concepts subsume each other, though. That is problematic.

**Comment by tomaszkam** [comment]
To be clear - I do not see problem with the ability to adapt the `executor` to `void-sender`, but the confusing part is that this adaptation is implicit, and making executors models both `senders` and `schedulers` implicitly. And the paper does not discuss, why all of this adaptation should be implicit.

I would much more prefer a design, where `executor` would implicitly model `scheduler`, and `schedule(e)` would be a syntax to transform `executor` into a `sender`, without having a special case in `execution::connect`.
Note that, current automatic wrapping, is inefficient due being partial, for example `execution::submit(e, r)` would fallback to default cause that uses `submit-receiver` and causes allocation, while submit can be implemented more efficiently for executor, as:
```
template<typename Receiver>
void executor_sender::submit(Receiver&& r)
{
    execution::execute(e, [r = std::forward<R>(r)] {
         try {
            execution::set_value(r);
        } catch {
            execution::set_error(r, std::current_exception());
        }
    });
}
```
(Because receivers are movable).

Similarly, I do not see (in paper) any use case explained, for `sender` being implicitly treated as `executor` via `execution::execute`. The problem with that wrapping is that it chooses one error handling strategy (terminate), while others may be possible, e.g. if passed `invocable` has `execption_ptr` overload, or receives `error_code` argument. I would suggest removing such implicit adaptation and eventually provide `as_receiver` as a library utility.

**Comment by ericniebler** [comment]
This is a sensible direction. Thanks for the feedback.
